src/_ert/forward_model_runner/reporting/event.py

```diff
@@ -175,21 +186,21 @@ def _job_handler(self, msg: Union[Start, Running, Exited]):
                 current_memory_usage=msg.memory_status.rss,
                 cpu_seconds=msg.memory_status.cpu_seconds,
             )
-            self._dump_event(event)
+            await self._dump_event(event)
 
-    def _finished_handler(self, _):
-        self._event_queue.put(Event._sentinel)
-        with self._timestamp_lock:
-            self._timeout_timestamp = datetime.now() + timedelta(
-                seconds=self._reporter_timeout
-            )
-        if self._event_publisher_thread.is_alive():
-            self._event_publisher_thread.join()
+    async def _finished_handler(self):
+        await self._event_queue.put(Event._sentinel)
+        self._timeout_timestamp = datetime.now() + timedelta(
+            seconds=self._reporter_timeout
+        )
 
-    def _checksum_handler(self, msg: Checksum):
+    async def _checksum_handler(self, msg: Checksum) -> None:
         fm_checksum = ForwardModelStepChecksum(
             ensemble=self._ens_id,
             real=self._real_id,
             checksums={msg.run_path: msg.data},
         )
-        self._dump_event(fm_checksum)
+        await self._dump_event(fm_checksum)
+
+    def cancel(self) -> None:
+        self._event_publishing_task.cancel()
```

Comment:
If this suppose to be a "blocking" operation then we should await to task to finish afterwards; ie.
```python
self._event_publishing_task.cancel()
try:
   await self._event_publishing_task
except asyncio.CancelledError:
 ...
```
```

Reply:
This also has to be synced as it is used in the eventloop's signal_handler

Reply:
right, I need to look deeper in the loop signal_handler :+1: